normandy/recipes/api/views.py

```diff
@@ -6,14 +6,15 @@
 from rest_framework.exceptions import NotFound
 from rest_framework.response import Response
 
+from normandy.base.api.mixins import RevisionMixin
 from normandy.base.api.permissions import AdminEnabled
 from normandy.base.api.renderers import JavaScriptRenderer
 from normandy.recipes.models import Action
 from normandy.recipes.api.permissions import NotInUse
 from normandy.recipes.api.serializers import ActionSerializer
 
 
-class ActionViewSet(viewsets.ModelViewSet):
+class ActionViewSet(RevisionMixin, viewsets.ModelViewSet):
```

Comment:
Doh. I misread that. I'm still concerned about the order of things. The update method here doesn't create a revision or become atomic until it calls super(), which in this case is fine. The mixin isn't as fool proof as I'd like, but that's probably fine.